### ISSUE TITLE
Accessibility and use of semantic HTML in the documentation

### ISSUE THREAD

**Issue by chcederquist** [orval-labs/orval#2294]
I've found the following accessibility issues in the documentation:

- [ ] `<aside>` is before the main content without a skip link [WCAG SC 2.4.1](https://www.w3.org/TR/WCAG21/#bypass-blocks)
- [ ] The main content is not wrapped in a `<main>`-tag [WCAG SC 1.3.1](https://www.w3.org/TR/WCAG21/#info-and-relationships)
- [ ] Headings are inconsistent. Some pages start with a `<h3>`, others with an `<h1>`  [WCAG SC 1.3.1](https://www.w3.org/TR/WCAG21/#info-and-relationships)
- [ ] The headings inside `<aside>` start at `<h4>`, but should preferably start at `<h2>`  [WCAG SC 1.3.1](https://www.w3.org/TR/WCAG21/#info-and-relationships)
- [ ] gray-400 is too light compared to the white background, causing insufficient color contrast [WCAG SC 1.4.3](https://www.w3.org/TR/WCAG21/#contrast-minimum)
- [ ] The header navigation should use `<nav>` [WCAG SC 1.3.1](https://www.w3.org/TR/WCAG21/#info-and-relationships)
- [ ] The navigation at the end of a documentation page should also use `<nav>` [WCAG SC 1.3.1](https://www.w3.org/TR/WCAG21/#info-and-relationships)

If you agree, I'd like to add a <main> tag to the main content, <nav> to the navigational mechanisms, use gray-500 instead of gray-400 where relevant and redo the heading levels to fit semantically, while keeping the proper sizing by using classes (e.g. <h2 class="h4">)

**Comment by melloware** [comment]
Totally agreed and assigned to you!

**Comment by melloware** [comment]
Can you add this page to your list: https://orval.dev/overview

Orval also generates with Fetch client now, Hono, MCP and more and that page only mentions Query + Axios and Angular.

**Comment by chcederquist** [comment]
I've added the Overview-page to the TODO list on the other issue
